app/src/main/java/com/example/criminalintentapp/CrimeListFragment.kt

```diff
@@ -1,41 +1,72 @@
 package com.example.criminalintentapp
 
+import android.content.Context
 import android.os.Bundle
 import android.util.Log
 import android.view.View
+import android.widget.TextView
+import androidx.core.view.isVisible
 import androidx.fragment.app.Fragment
 import androidx.lifecycle.Observer
 import androidx.lifecycle.ViewModelProvider
 import androidx.recyclerview.widget.LinearLayoutManager
 import androidx.recyclerview.widget.RecyclerView
+import java.util.UUID
 
 class CrimeListFragment : Fragment(R.layout.fragment_crime_list) {
 
+    interface Callbacks {
+        fun onCrimeSelected(crimeId: UUID)
+    }
+
+    var callbacks: Callbacks? = null
+
     private lateinit var crimeRecyclerView: RecyclerView
+    private lateinit var emptyListTextView: TextView
     private var adapter: CrimeAdapter = CrimeAdapter(emptyList())
 
     private val crimeListViewModel: CrimeListViewModel by lazy {
         ViewModelProvider(this).get(CrimeListViewModel::class.java)
     }
 
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        callbacks = context as Callbacks?
+    }
+
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         crimeListViewModel.crimesListLiveData.observe(
             viewLifecycleOwner,
             Observer { crimes ->
+                if (crimes.isEmpty()) {
+                    emptyListTextView.isVisible = true
```

Comment:
The app crashes here:
```
  kotlin.UninitializedPropertyAccessException: lateinit property emptyListTextView has not been initialized
        at com.example.criminalintentapp.CrimeListFragment.onViewCreated$lambda-0(CrimeListFragment.kt:43)
```
You need to make sure that all the views are set up before using them